vnext/Microsoft.ReactNative/Views/FocusZoneViewManager.cpp

```diff
@@ -0,0 +1,151 @@
+  m_panelLosingFocusRevoker =
+      panel.LosingFocus(winrt::auto_revoke, [=](const auto &sender, const auto &e) { OnLosingFocus(sender, e); });
+}
+
+void FocusZoneViewShadowNode::updateProperties(winrt::Microsoft::ReactNative::JSValueObject &props) {
+  for (const auto &pair : props) {
+    const std::string &propertyName = pair.first;
+    const auto &propertyValue = pair.second;
+
+    if (propertyName == "focusDirection") {
+      if (propertyValue.Type() == winrt::Microsoft::ReactNative::JSValueType::String) {
+        m_isHorizontal = propertyValue.AsString() == "horizontal";
+      }
+    }
+  }
+}
+
+void FocusZoneViewShadowNode::OnKeyDown(const winrt::IInspectable &sender, const xaml::Input::KeyRoutedEventArgs &e) {
+  auto senderFrameworkElement = sender.try_as<xaml::FrameworkElement>();
+  auto xamlRoot = senderFrameworkElement.XamlRoot();
+  auto focusedElement = xaml::Input::FocusManager::GetFocusedElement(xamlRoot).try_as<xaml::DependencyObject>();
+  auto findNextElementOptions = xaml::Input::FindNextElementOptions();
+  findNextElementOptions.SearchRoot(xamlRoot.Content());
+
+  auto previousKey = m_isHorizontal ? winrt::system::VirtualKey::Left : winrt::system::VirtualKey::Up;
+  auto nextKey = m_isHorizontal ? winrt::system::VirtualKey::Right : winrt::system::VirtualKey::Down;
+
+  if (e.Key() == previousKey) {
+    auto firstFocusableElement =
+        xaml::Input::FocusManager::FindFirstFocusableElement(senderFrameworkElement).try_as<xaml::DependencyObject>();
+    auto isFirstElementFocused = firstFocusableElement == focusedElement;
+    if (!isFirstElementFocused) {
+      xaml::Input::FocusManager::TryMoveFocus(xaml::Input::FocusNavigationDirection::Previous, findNextElementOptions);
+      e.Handled(true);
+    }
+  } else if (e.Key() == nextKey) {
+    auto lastFocusableElement =
+        xaml::Input::FocusManager::FindLastFocusableElement(senderFrameworkElement).try_as<winrt::DependencyObject>();
+    auto isLastElementFocused = lastFocusableElement == focusedElement;
+    if (!isLastElementFocused) {
+      xaml::Input::FocusManager::TryMoveFocus(xaml::Input::FocusNavigationDirection::Next, findNextElementOptions);
+      e.Handled(true);
+    }
+  } else if (e.Key() == winrt::system::VirtualKey::Tab) {
+    if (!react::uwp::IsXamlIsland()) {
+      auto const &coreWindow = winrt::CoreWindow::GetForCurrentThread();
+      auto isShiftDown = KeyboardHelper::IsModifiedKeyPressed(coreWindow, winrt::system::VirtualKey::Shift);
+
+      winrt::Point anchorTopLeft = winrt::Point(0, 0);
+      xaml::Media::GeneralTransform transform = senderFrameworkElement.TransformToVisual(xamlRoot.Content());
+      winrt::Point anchorTopLeftConverted = transform.TransformPoint(anchorTopLeft);
+      auto exclusionRect = winrt::Rect(
+          anchorTopLeftConverted.X,
+          anchorTopLeftConverted.Y,
+          static_cast<float>(senderFrameworkElement.Width()),
+          static_cast<float>(senderFrameworkElement.Height()));
```

Comment:
these can be NaN if this happens before layout, use `ActualWidth` and `ActualHeight` instead

Reply:
Got lost after applying the change to ScrollViewManager.. fixing in next iteration.